src/js/administration/api.ts

```diff
@@ -0,0 +1,66 @@
+import { Request } from "../app/request";
+/**
+ * Fetch the current settings from the server.
+ *
+ * @returns - A promise resolving to the current server settings.
+ */
+export function fetchSettings() {
+    return Request.get("/settings").then(response => {
+        return response.body;
+    });
+}
+
+/**
+ * Fetch the current settings from the server.
+ *
+ * @returns - A promise resolving to the complete response, including the servers settings.
+ */
+export function legacyFetchSettings() {
+    return Request.get("/settings");
+}
+
+/**
+ * Update the current settings on the server.
+ *
+ * @returns - A promise resolving to the complete response containing the updated settings.
+ */
+export function updateSettings(update) {
+    return Request.patch("/settings").send(update);
+}
+
+/**
+ * Fetch a list of valid administrator roles
+ *
+ * @returns - A promise resolving to the list of valid administrator roles
+ */
+export function fetchAdministratorRoles() {
+    return Request.get("/admin/roles").then(response => response.body);
+}
```

Comment:
You are missing return types here, but you included them in other `api.ts` modules.